Source/FuseJS/Tests/BundleTests.uno

```diff
@@ -0,0 +1,160 @@
+using Uno;
+using Uno.IO;
+using Uno.Collections;
+using Uno.Testing;
+using Uno.Threading;
+using Uno.Text;
+using FuseTest;
+using Uno.Compiler.ExportTargetInterop;
+
+namespace FuseJS.Test
+{
+	/*
+		File path stuff cribbed from Fuse.FileSystem. That stuff should probably be made public/simpler for dogfooding.
+	*/
+
+	[extern(DOTNET) DotNetType("System.Reflection.Assembly")]
+	internal extern(DOTNET) abstract class BclAssembly
+	{
+		public extern static BclAssembly GetExecutingAssembly();
+		public extern string Location { get; }
+	}
+
+	class Utils
+	{	
+		sealed class Nothing : object
+		{
+			private Nothing() {}
+		}
+
+		private class Box<T>
+		{
+			T _value;
+
+			public T Value { get { return _value; } }
+
+			public void Deliver(T value)
+			{
+				_value  = value;
+			}
+		}
+		
+		static string _cacheDirectory;
+
+		public static T WaitForSuccessAndGetValue<T>(Future<T> future)
+		{
+			future.Wait();
+			var resultBox = new Box<T>();
+			var exceptionBox = new Box<Exception>();
+			future.Then(resultBox.Deliver, exceptionBox.Deliver);
+			Assert.AreEqual(exceptionBox.Value, null, "Threw exception from future: " + exceptionBox.Value);
+			if (!(typeof(T) == typeof(Nothing)))
+				Assert.AreNotEqual(resultBox.Value, null);
+			return resultBox.Value;
+		}
+
+		static string GetAppBaseDirectory()
+		{
+			string exeDir;
+			if defined(DOTNET)
+			{
+				exeDir = Path.GetDirectoryName(Path.GetFullPath(BclAssembly.GetExecutingAssembly().Location));
+			}
+			else
+			{
+				exeDir = Path.GetDirectoryName(Path.GetFullPath(Environment.GetCommandLineArgs()[0]));
+			}
+			return exeDir;
+		}
+
+		public static string GetCacheDirectory()
+		{
+			return GetOrCreateDirectory(ref _cacheDirectory, "fs_cache");
+		}
+
+
+		static string GetOrCreateDirectory(ref string path, string name)
+		{
+			if (path == null)
+			{
+				// This should be ok wrt. race conditions as it won't matter if directory
+				// is created more than one time.. (no locks should be required)
+				var baseDir = GetAppBaseDirectory();
+				path = NormalizePath(Path.Combine(baseDir, name));
+				Directory.CreateDirectory(path);
+			}
+			return path;
+		}
+
+		public static string NormalizePath(string path)
+		{
+			if (Path.DirectorySeparatorChar == '\\')
+				return path.Replace('\\', '/');
+			return path;
+		}
+	}
+
+	
+
+	/*
+		End theft
+	*/
+
+	
+	public class BundleTests : TestBase
+	{
+
+		[Test]
+		public void TestList()
+		{
+			var result = Utils.WaitForSuccessAndGetValue(FuseJS.Bundle.GetList());
+			Assert.IsTrue(result.Count()>0, "No bundles found");
```

Comment:
This API should really just return a list of every bundled file, no matter where it came from. As long as it returns a list of at least 1 file, the test is sound IMO.